.github/workflows/test_geopandas.yml

```
@@ -0,0 +1,45 @@
+name: Geopandas tests
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+
+jobs:
+  run:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout Folium
+      uses: actions/checkout@v4
+
+    - name: Setup Micromamba env
+      uses: mamba-org/setup-micromamba@v2
+      with:
+        environment-name: TEST
+        create-args: >-
+          python=3
+          --file requirements.txt
+
+    - name: Checkout Geopandas
+      uses: actions/checkout@v4
+      with:
+        repository: geopandas/geopandas
+        ref: main
+        path: geopandas  # Checkout into a subdirectory
+
+    - name: Install Geopandas dev dependencies
+      run: |
+        cd geopandas
+        pip install -r requirements-dev.txt
+
+    - name: Install folium from source
+      shell: bash -l {0}
+      run: |
+        python -m pip install -e . --no-deps --force-reinstall
+
+    - name: Run Geopandas tests
+      run: |
+        cd geopandas
+        pytest
```

```suggestion
        pytest geopandas/tests/test_explore.py
```

This is the only relevant file where we interact with folium in our CI. No need to run the other.

---

Thanks for the suggestion Martin!